spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsOutboundGateway.java

```diff
@@ -560,6 +564,16 @@ protected void doInit() {
 				setContainerProperties(container);
 				container.afterPropertiesSet();
 				this.replyContainer = container;
+				if (isAsyncReplySupported() && this.correlationKey == null) {
+					logger.warn("'async=true' requires a correlationKey; ignored");
```

Comment:
Ah! I know. Just because it doesn't see that the `logger` is from the super class :smile: